src/VisualStudio/Core/Def/Progression/GraphQueries/SearchGraphQuery.cs

```diff
@@ -6,46 +6,53 @@
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Progression;
+
+internal sealed partial class SearchGraphQuery(
+    string searchPattern,
+    NavigateToSearchScope searchScope,
+    IThreadingContext threadingContext,
+    IAsynchronousOperationListener asyncListener) : IGraphQuery
 {
-    internal sealed partial class SearchGraphQuery : IGraphQuery
+    public async Task<GraphBuilder> GetGraphAsync(Solution solution, IGraphContext context, CancellationToken cancellationToken)
     {
-        private readonly string _searchPattern;
-        private readonly NavigateToSearchScope _searchScope;
-        private readonly IThreadingContext _threadingContext;
-        private readonly IAsynchronousOperationListener _asyncListener;
-
-        public SearchGraphQuery(
-            string searchPattern,
-            NavigateToSearchScope searchScope,
-            IThreadingContext threadingContext,
-            IAsynchronousOperationListener asyncListener)
-        {
-            _threadingContext = threadingContext;
-            _asyncListener = asyncListener;
-            _searchPattern = searchPattern;
-            _searchScope = searchScope;
-        }
-
-        public async Task<GraphBuilder> GetGraphAsync(Solution solution, IGraphContext context, CancellationToken cancellationToken)
-        {
-            var graphBuilder = await GraphBuilder.CreateForInputNodesAsync(solution, context.InputNodes, cancellationToken).ConfigureAwait(false);
-            var callback = new ProgressionNavigateToSearchCallback(context, graphBuilder);
-            var searcher = NavigateToSearcher.Create(
-                solution,
-                _asyncListener,
-                callback,
-                _searchPattern,
-                NavigateToUtilities.GetKindsProvided(solution),
-                _threadingContext.DisposalToken);
-
-            await searcher.SearchAsync(searchCurrentDocument: false, _searchScope, cancellationToken).ConfigureAwait(false);
-
-            return graphBuilder;
-        }
+        var graphBuilder = await GraphBuilder.CreateForInputNodesAsync(solution, context.InputNodes, cancellationToken).ConfigureAwait(false);
+        var callback = new ProgressionNavigateToSearchCallback(context, graphBuilder);
+
+        // We have a specialized host for progression vs normal nav-to.  Progression itself will tell the client if
+        // the project is fully loaded or not.  But after that point, the client will be considered fully loaded and
+        // results should reflect that.  So we create a host here that will always give complete results once the
+        // solution is loaded and not give cached/incomplete results at that point.
+        var statusService = solution.Services.GetRequiredService<IWorkspaceStatusService>();
+        var isFullyLoaded = await statusService.IsFullyLoadedAsync(cancellationToken).ConfigureAwait(false);
+        var host = new SearchGraphQueryNavigateToSearchHost(isFullyLoaded);
```

Comment:
Progression can send us queries that are "live" that we'll update as solution changes happen -- does that end up working for text searches? I'd imagine that once the solution is fully loaded we'll end up refreshing and the user ends up with complete results. It might be very _eventually_ complete, but it would complete, right?

(I have a separate question for whether we also do that for regular Navigate To but I'm a bit confused on the semantics there.)